tests/tests/test_guest_memory.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import time
+import json
+import subprocess
+
+import Qemu
+from common import *
+
+MEMORY_FILE='memory.bin'
+CHECK_EXEC='export INCLEAR=6150079244'
+
+def check_for_string(filename, needle):
+    cs = subprocess.run(['sudo', 'grep', needle, filename], check=True, capture_output=True)
```

Comment:
with `check=True`, the call will throw an exception if the string is not present in the file, that will make the test `test_guest_memory_confidentiality_tdx` fails and we do not want that, do we ?

Reply:
Yes, thanks for catching that.

Reply:
fixed in https://github.com/canonical/tdx/pull/235/commits/ea47d4ddcd72bba77caaf26e041bcc8cf3132d0e